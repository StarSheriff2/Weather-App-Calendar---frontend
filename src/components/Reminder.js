import React from 'react';
import PropTypes from 'prop-types';
import WeatherForecast from './WeatherForecast';

const Reminder = ({ reminder }) => {
  const {
    description, city,
    date, time, location_coordinates: coordinates,
  } = reminder;

  const getDateDiff = (first, second) => Math.round((second - first) / (1000 * 60 * 60 * 24));

  // Compute difference in min between current t and reminder t
  const timeNow = new Date();
  const reminderTime = new Date(`${date}T${time}:00-06:00`);
  const diffMs = +timeNow - +reminderTime;
  const diffMins = Math.floor((diffMs / 1000) / 60);

  // Compute difference in days
  const dateDiff = getDateDiff(new Date(), reminderTime);

  let weatherContent = null;
  if (dateDiff < 0) {
    weatherContent = (
      <div>
        <i className="fas fa-exclamation-circle text-muted" />
      </div>
    );
  } else if (dateDiff > 7) {
    weatherContent = (
      <div>
        <i className="fas fa-exclamation-circle" />
      </div>
    );
  } else {
    weatherContent = (
      <WeatherForecast
        coordinates={coordinates}
        dateTime={reminderTime}
        dateDiff={dateDiff}
      />
    );
  }

  return (
    <tr
      className={(diffMins <= 20 && diffMins >= 0) ? 'table-row active-reminder' : 'table-row'}
    >
      <td className={(diffMins <= 20 && diffMins >= 0) ? 'glow align-middle' : 'table-cell align-middle'}><small>{time}</small></td>
      <td className="align-middle">{description}</td>
      <td className="align-middle">{city.split(', ')[0]}</td>
      <td className="align-middle">
        {weatherContent}
      </td>
    </tr>
  );
};

Reminder.propTypes = {
  reminder: PropTypes.shape({
    description: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    time: PropTypes.string.isRequired,
    location_coordinates: PropTypes.string.isRequired,
  }).isRequired,
};

export default Reminder;
