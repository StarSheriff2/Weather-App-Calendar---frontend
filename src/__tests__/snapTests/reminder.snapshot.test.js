import React from 'react';
import { cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderWithRedux from '../../tests/test-utils';
import Reminder from '../../components/Reminder';

afterEach(cleanup);

describe('Reminder', () => {
  test('Renders the Component', async () => {
    const reminderProp = {
      description: 'quod consequatur',
      date: '2022-02-03',
      time: '05:39',
      city: 'Connville',
      location_coordinates: '31.438327740769708, -69.93035045104288',
    };

    const reminder = await renderWithRedux(
      <Router>
        <Reminder reminder={reminderProp} />
      </Router>,
    ).toJSON();

    expect(reminder).toMatchSnapshot();
  });
});
