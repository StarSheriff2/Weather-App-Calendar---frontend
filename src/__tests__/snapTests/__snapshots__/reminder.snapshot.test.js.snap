// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reminder Renders the Component 1`] = `
<tr
  className="table-row"
>
  <td
    className="table-cell align-middle"
  >
    <small>
      05:39
    </small>
  </td>
  <td
    className="align-middle"
  >
    quod consequatur
  </td>
  <td
    className="align-middle"
  >
    Connville
  </td>
  <td
    className="align-middle"
  >
    <span
      className="spinner-border spinner-border-sm"
    />
  </td>
</tr>
`;
